src/librustc_mir/transform/check_unsafety.rs

```diff
@@ -536,8 +536,10 @@ fn unsafety_check_result(tcx: TyCtxt<'_>, def_id: DefId) -> UnsafetyCheckResult
     };
     let mut checker = UnsafetyChecker::new(
         const_context, min_const_fn,
-        body, source_scope_local_data, tcx, param_env);
-    checker.visit_body(body);
+        body_cache, source_scope_local_data, tcx, param_env);
+    let mut cache = body_cache.cache().clone();
+    let read_only_cache = ReadOnlyBodyCache::from_external_cache(&mut cache, body_cache.body());
```

Comment:
I'm not really sure how to avoid this. `mir_built` is behind a `Steal`, and this was previously a borrow. There's no way for me to update its existing cache here. If we can somehow work with that or come up with a different pattern, then we can make `Cache` private to the rustc/mir module. Otherwise I need to make it public and expose the `cache` fn on `BodyCache` so it can be read above to make a clone.

Reply:
@oli-obk this is the comment I was referring to in Zulip